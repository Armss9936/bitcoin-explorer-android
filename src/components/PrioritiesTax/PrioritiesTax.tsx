import { View, Text, StyleSheet } from "react-native";
import React, { useContext } from "react";
import SmallBoxInfo from "../SmallBoxInfo/SmallBoxInfo";
import useAppDataStore from "src/context/DataProvider";
import { I18nContext } from "src/context/LocaleProvider";

export default function PrioritiesTax() {
  const { fees } = useAppDataStore();
  const i18nContext = useContext(I18nContext);

  return (
    <>
      <View style={styles.taxContainerPrioritiesTaxText}>
        <Text style={styles.taxPrioritiesTaxText}>
          {i18nContext.t("low_priority")}
        </Text>
        <Text style={styles.taxPrioritiesTaxText}>
          {i18nContext.t("medium_priority")}
        </Text>
        <Text style={styles.taxPrioritiesTaxText}>
          {i18nContext.t("high_priority")}
        </Text>
      </View>
      <SmallBoxInfo boxesInfos={fees} />
    </>
  );
}

const styles = StyleSheet.create({
  taxContainerPrioritiesTaxText: {
    width: "100%",
    flexDirection: "row",
    justifyContent: "space-around",
    backgroundColor: "#1d2133",
    padding: 16,
    borderRadius: 8,
  },
  taxPrioritiesTaxText: {
    color: "#FFF",
    fontSize: 12,
  },
});
