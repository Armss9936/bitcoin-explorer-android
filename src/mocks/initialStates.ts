import { TransactionType } from "src/components/Modal/types";

const timeAgoInitialState = {
  hour: 0,
  minutes: 0,
  day: 0,
  month: 0,
  year: 0,
};

export const initialStateBlocks = [
  {
    blockHeight: 1,
    satPerVbyte: "1",
    size: "1",
    transactions: 1,
    timeAgo: timeAgoInitialState,
  },
  {
    blockHeight: 2,
    satPerVbyte: "1",
    size: "1",
    transactions: 1,
    timeAgo: timeAgoInitialState,
  },
  {
    blockHeight: 3,
    satPerVbyte: "1",
    size: "1",
    transactions: 1,
    timeAgo: timeAgoInitialState,
  },
  {
    blockHeight: 4,
    satPerVbyte: "1",
    size: "1",
    transactions: 1,
    timeAgo: timeAgoInitialState,
  },
];

export const initialStateTransaction: TransactionType = {
  transactionId: " ",
  size: 0,
  fee: 0,
  inputTransactions: [
    {
      prevout: {
        scriptpubkey_address: " ",
        value: 0,
      },
    },
  ],
  outputTransactions: [
    {
      scriptpubkey_address: " ",
      value: 0,
    },
  ],
  statusTransaction: undefined,
};
