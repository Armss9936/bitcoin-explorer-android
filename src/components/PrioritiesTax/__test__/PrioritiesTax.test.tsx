/* eslint-disable import/no-extraneous-dependencies */
import { I18n } from "i18n-js";
import renderer from "react-test-renderer";
import { initialStateFees } from "src/__mocks__/initialStates";
import PrioritiesTax from "../PrioritiesTax";
import "expo-localization"; // Importa para garantir que o mock estÃ¡ sendo usado

// Mocking the i18n function
jest.mock("i18n-js", () => {
  const actualI18nJs = jest.requireActual("i18n-js");

  return {
    __esModule: true, // Ensures that the default export works as expected
    default: jest.fn(() => ({
      t: jest.fn((key) => key),
      currentLocale: "en",
      locale: "en",
      defaultLocale: "en",
      enableFallback: true,
    })),
    I18n: actualI18nJs.I18n, // Export the actual I18n constructor
  };
});

// _i18nJs.I18n is not a constructor - solving this error

const localization = {
  t: (value: string) => value,
  currentLocale: "en",
};

describe("PrioritiesTax", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <PrioritiesTax
          fees={initialStateFees}
          localization={localization as unknown as I18n}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
