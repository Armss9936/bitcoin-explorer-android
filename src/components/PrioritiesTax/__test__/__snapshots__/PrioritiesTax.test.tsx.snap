// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrioritiesTax renders correctly 1`] = `
[
  <View
    style={
      {
        "backgroundColor": "#1d2133",
        "borderRadius": 8,
        "flexDirection": "row",
        "justifyContent": "space-around",
        "padding": 16,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#FFF",
          "fontSize": 12,
        }
      }
    >
      low_priority
    </Text>
    <Text
      style={
        {
          "color": "#FFF",
          "fontSize": 12,
        }
      }
    >
      medium_priority
    </Text>
    <Text
      style={
        {
          "color": "#FFF",
          "fontSize": 12,
        }
      }
    >
      high_priority
    </Text>
  </View>,
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-around",
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#1d2133",
          "borderRadius": 7,
          "paddingHorizontal": 8,
          "paddingVertical": 16,
          "textAlign": "center",
          "width": 100,
        }
      }
    >
      <Text
        style={
          {
            "color": "#FFF",
            "fontSize": 14,
          }
        }
      >
         
         sat/vB
      </Text>
      <Text
        style={
          {
            "color": "#DF7800",
          }
        }
      >
         
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#1d2133",
          "borderRadius": 7,
          "paddingHorizontal": 8,
          "paddingVertical": 16,
          "textAlign": "center",
          "width": 100,
        }
      }
    >
      <Text
        style={
          {
            "color": "#FFF",
            "fontSize": 14,
          }
        }
      >
         
         sat/vB
      </Text>
      <Text
        style={
          {
            "color": "#DF7800",
          }
        }
      >
         
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#1d2133",
          "borderRadius": 7,
          "paddingHorizontal": 8,
          "paddingVertical": 16,
          "textAlign": "center",
          "width": 100,
        }
      }
    >
      <Text
        style={
          {
            "color": "#FFF",
            "fontSize": 14,
          }
        }
      >
         
         sat/vB
      </Text>
      <Text
        style={
          {
            "color": "#DF7800",
          }
        }
      >
         
      </Text>
    </View>
  </View>,
]
`;
